ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/plugin/ExecutionEnvironmentSection.java

```diff
@@ -127,7 +133,12 @@ public Object[] getElements(Object inputElement) {
 	}
 
 	public ExecutionEnvironmentSection(PDEFormPage page, Composite parent) {
-		super(page, parent, Section.DESCRIPTION, new String[] {PDEUIMessages.RequiredExecutionEnvironmentSection_add, PDEUIMessages.RequiredExecutionEnvironmentSection_remove, PDEUIMessages.RequiredExecutionEnvironmentSection_up, PDEUIMessages.RequiredExecutionEnvironmentSection_down});
+		super(page, parent, Section.DESCRIPTION,
+				new String[] { PDEUIMessages.RequiredExecutionEnvironmentSection_add,
+						PDEUIMessages.RequiredExecutionEnvironmentSection_upgrade,
+						PDEUIMessages.RequiredExecutionEnvironmentSection_remove,
+						PDEUIMessages.RequiredExecutionEnvironmentSection_up,
+						PDEUIMessages.RequiredExecutionEnvironmentSection_down });
```

Comment:
Since `Upgrade` does more than Add/Remove/Up/Down I think it should be either first or last to have the existing buttons grouped.
```suggestion
				new String[] {  PDEUIMessages.RequiredExecutionEnvironmentSection_upgrade,
						PDEUIMessages.RequiredExecutionEnvironmentSection_add,
						PDEUIMessages.RequiredExecutionEnvironmentSection_remove,
						PDEUIMessages.RequiredExecutionEnvironmentSection_up,
						PDEUIMessages.RequiredExecutionEnvironmentSection_down });
```